server/internal/interface/grpc/handlers/arkservice.go

```diff
@@ -156,6 +156,45 @@ func (h *handler) RegisterIntent(
 	}, nil
 }
 
+func (h *handler) DeleteIntent(
+	ctx context.Context, req *arkv1.DeleteIntentRequest,
+) (*arkv1.DeleteIntentResponse, error) {
+	requestID := req.GetRequestId()
+	bip322Signature := req.GetBip322Signature()
+
+	if requestID != "" {
```

Comment:
AH i see it's already in the code as I described.  Just needs the proto to use the oneof style?